src/shared/utilities/uniqueId.ts

```diff
@@ -1,8 +1,10 @@
+import { useMemo } from 'react';
 /**
  * Generates a simple unique id
  * @param prefix The sting prefix
  */
 export const uniqueId = ((): ((prefix: string) => string) => {
   let counter: number = 0;
-  return (prefix: string): string => `${prefix}${++counter}`;
+  return (prefix: string): string =>
+    useMemo(() => `${prefix}${++counter}`, [prefix]);
```

Comment:
won't this result in all inputs in the app lifecycle having the same "unique" id?